bundles/org.openhab.binding.energenie/src/main/java/org/openhab/binding/energenie/internal/handler/EnergeniePWMHandler.java

```diff
@@ -0,0 +1,200 @@
+                final String slicedResponseTmp = loginResponseString.substring(findState + start, findState + stop);
+                logger.trace("transformed state response = {}", slicedResponseTmp);
+                final String[] slicedResponse = slicedResponseTmp.split(";");
+                logger.trace("transformed state response = {} - {}", slicedResponse[0], slicedResponse[1]);
+                final double value;
+
+                if (Double.parseDouble(slicedResponse[0]) / 1 == Double.parseDouble(slicedResponse[0])) {
+                    value = Double.parseDouble(slicedResponse[0]) / divisor;
+                } else {
+                    value = -1.0;
+                }
+                return QuantityType.valueOf(value, unit);
+            } else {
+                logger.trace("searchstring '{} not found", stateResponseSearch);
+                return UnDefType.UNDEF;
+            }
+        }
+    }
+
+    private final Logger logger = LoggerFactory.getLogger(EnergeniePWMHandler.class);
+
+    private static final int HTTP_TIMEOUT_MILLISECONDS = 6000;
+
+    private @Nullable EnergenieConfiguration config;
+
+    private String host = "";
+    private String password = "";
+    private int refreshInterval;
+
+    @Nullable
+    private ScheduledFuture<?> refreshJob;
+
+    public EnergeniePWMHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        // PWM devices don't support any commands, just value reading
+    }
+
+    @Override
+    public void initialize() {
+        EnergenieConfiguration config = getConfigAs(EnergenieConfiguration.class);
+
+        this.config = config;
+
+        if (config.host != null && config.password != null) {
+            host = config.host;
+            password = config.password;
+            refreshInterval = EnergenieConfiguration.DEFAULT_REFRESH_INTERVAL;
+            logger.debug("Initializing EnergeniePWMHandler for Host '{}'", host);
+            updateStatus(ThingStatus.ONLINE);
+            onUpdate();
+        } else {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    "Can not access device , IP-Address or password not set");
+        }
+    }
+
+    @Override
+    public void dispose() {
+        logger.debug("EnergeniePWMHandler disposed.");
+        if (refreshJob != null && !refreshJob.isCancelled()) {
+            refreshJob.cancel(true);
+            refreshJob = null;
+        }
+    }
+
+    public void getState() {
+        String url = "http://" + host + "/login.html";
+        String urlParameters = "pw=" + password;
+        InputStream urlContent = new ByteArrayInputStream(urlParameters.getBytes(StandardCharsets.UTF_8));
+        String loginResponseString = null;
+
+        try {
+            logger.trace("sendlogin to {} with password {}", host, password);
```

Comment:
Please do not log passwords. Maybe log the length of the password if you want to make sure that the password is correctly set

Reply:
Done